ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,493 @@
+
+Output image will have the filename of <inputname>_<suffix>.fits.""")
+
+    def redo(self, *args):
+        """Generate listing of images that user can save."""
+        if not self.gui_up:
+            return
+
+        #mod_only = self.settings.get('modified_only', True)
+        mod_only = self.w.modified_only.get_state()
+        treedict = Bunch.caselessDict()
+        self.treeview.clear()
+        self.w.status.set_text('')
+
+        channel = self.fv.get_channelInfo(self.chname)
+        if channel is None:
+            return
+
+        # Only list modified images for saving. Scanning Datasrc is enough.
+        if mod_only:
+            func = self.keys_from_datasrc
+
+        # List all images in the channel.
+        else:
+            func = channel.get_image_names
+
+        # Extract info for listing and saving
+        for key in func():
+            iminfo = channel.get_image_info(key)
+            path = iminfo.get('path')
+            idx = iminfo.get('idx')
+            t = iminfo.get('time_modified')
+
+            if path is None:  # Special handling for generated buffer, eg mosaic
+                infile = key
+                is_fits = True
+            else:
+                infile = os.path.basename(path)
+                infile_ext = os.path.splitext(path)[1]
+                infile_ext = infile_ext.lower()
+                is_fits = False
+
+                if 'fit' in infile_ext:
+                    is_fits = True
+
+            # Only list FITS files unless it is Ginga generated buffer
+            if not is_fits:
+                continue
+
+            # Only list modified buffers
+            if mod_only and t is None:
+                continue
+
+            # More than one ext modified, append to existing entry
+            if infile in treedict:
+                if t is not None:
+                    treedict[infile].extlist.add(idx)
+                    elist = sorted(treedict[infile].extlist)
+                    treedict[infile].MODEXT = ';'.join(
+                        map(self._format_extname, elist))
+
+            # Add new entry
+            else:
+                if t is None:
+                    s = ''
+                    extlist = set()
+                else:
+                    s = self._format_extname(idx)
+                    extlist = set([idx])
+                treedict[infile] = Bunch.Bunch(
+                    IMAGE=infile, MODEXT=s, extlist=extlist, path=path)
+
+        self.treeview.set_tree(treedict)
+
+        # Resize column widths
+        n_rows = len(treedict)
+        if n_rows == 0:
+            self.w.status.set_text('Nothing available for saving')
+        elif n_rows < self.settings.get('max_rows_for_col_resize', 5000):
+            self.treeview.set_optimal_column_widths()
+            self.logger.debug('Resized columns for {0} row(s)'.format(n_rows))
+
+        return
```

Comment:
The `return` can be removed as it does not really do anything. Probably I forgot to delete it.